drivers/xen/xenbus/xs_comms.c

```diff
@@ -612,13 +624,30 @@ int xs_comms_init(void)
 		      HYPERVISOR_start_info->store_mfn;
 #else
 #error "Unsupported architecture"
+#endif
+#else /* PVH && X86_64 */
+	UK_ASSERT(hvm_get_parameter(HVM_PARAM_STORE_EVTCHN, &evtchn) == 0);
+	UK_ASSERT(hvm_get_parameter(HVM_PARAM_STORE_PFN, &store_pfn) == 0);
+
+	pt = ukplat_pt_get_active();
+	rc = ukplat_page_map(pt, store_pfn << PAGE_SHIFT, store_pfn << PAGE_SHIFT, 1, PAGE_ATTR_PROT_RW, 0);
```

Comment:
the xenstore pfn is actually guest RAM (and <4G), which should be already mapped (many places says that the first 4GB are identity mapped)
so it will actually unmap it to remap it afterward at the same place which is not very useful